build_tools/release_automation.py

```diff
@@ -0,0 +1,262 @@
+    if r.status_code != 200:
+        print("Failure of zenodo connection check. Terminating.")
+        sys.exit(1)
+
+
+    #create empty upload
+    headers = {"Content-Type": "application/json"}
+    r = requests.post(zenodo_url+'/api/deposit/depositions',
+                   params={'access_token':zenodo_api_key}, json={},
+                   headers=headers)
+    if r.status_code != 201:
+        print("Failure of zenodo record creation. Terminating")
+        sys.exit(1)
+    else:
+        newDOI = r.json()['metadata']['prereserve_doi']['doi']
+        print("Empty record created with DOI:"+newDOI)
+        #print(r.json())
+
+    #populate record
+    deposition_id = r.json()['id']
+    r = requests.put(zenodo_url+'/api/deposit/depositions/%s' % deposition_id,
+                  params={'access_token': zenodo_api_key}, data=json.dumps(sasview_data),
+                  headers=headers)
+    if r.status_code != 200:
+        print("Failure to set metadata on record. Terminating")
+        sys.exit(1)
+    else:
+        print("Record metadata for DOI %s updated" % newDOI)
+
+
+    #next, manually add necessary files from Github and press the publish button
+
+    print("Record has been created on Zenodo with DOI: "+newDOI+"\nNext:\n- Add DOI to appropriate places in SasView code\n- "
+                                                            "Make GitHub release\n - Download all release files from GitHub "
+                                                            "and attach to Zenodo release record.\n- Publish Zenodo record")
+    return newDOI
+
+def update_sasview_init(version, doi):
+    """
+
+    :return:
+    """
+
+    init_file = os.path.join('sasview','src', 'sas', 'sasview', '__init__.py')
+    output_lines = []
+    year = datetime.datetime.now().year
+    with open(init_file, 'r') as f:
+        for line in f.readlines():
+            if line[:11] == '__version__':
+               output_lines.append('__version__ = \"'+version+'\"\n')
+            elif line[:7] == '__DOI__' :
+                output_lines.append('__DOI__ = \"Zenodo, ' + str(doi) + '\"\n')
+            elif line[:16] == '__release_date__':
+                output_lines.append('__release_date__ = \"' + str(year) + '\"\n')
+            else:
+                output_lines.append(line)
+
+    with open(init_file, 'w') as f:
+        f.writelines(output_lines)
+
+
+def update_sasmodels_init(version):
+    """
+
+    :param version:
+    :return:
+    """
+    init_file = os.path.join('sasmodels', 'sasmodels', '__init__.py')
+    output_lines = []
+    with open(init_file, 'r') as f:
+        for line in f.readlines():
+            if line[:11] == '__version__':
+                output_lines.append('__version__ = \"' + version + '\"\n')
+            else:
+                output_lines.append(line)
+    with open(init_file, 'w') as f:
+        f.writelines(output_lines)
+
+def update_license(license_file, license_line, line_to_update):
```

Comment:
This is very minor issue but the method is not at all license specific and would use a rename.
`upate_file` is much more fitting I would say.